### PR TITLE
Dockerize it

### PR DESCRIPTION
**Pull request by cpwallah**

### CODE REVIEW

File: ninjareact/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+# Install any needed packages specified in requirements.txt
+RUN pip install --no-cache-dir -r requirements.txt
+
+# Copy the rest of the application code into the container
+COPY . /app/
+
+# Run database migrations, collect static files, and create superuser if needed
+RUN python manage.py migrate
+RUN python manage.py collectstatic --noinput
```
Comment (joefutrelle) [inline ninjareact/Dockerfile:18]: it doesn't make sense to run these commands at build time--these steps need to be taken at deploy time.

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+      - app_network
+
+  frontend:
+    build:
+      context: ./ninjareact/frontend
+    container_name: react_frontend
+    ports:
+      - "3000:3000"
+    networks:
+      - app_network
```
Comment (joefutrelle) [inline docker-compose.yml:23]: There should not be a frontend service--the frontend is served by Django.

File: ninjareact/frontend/Dockerfile
Comment (joefutrelle) [inline ninjareact/frontend/Dockerfile:1]: There should be no frontend service (see comment on compose file)

File: docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,27 @@
+services:
+  backend:
+    build:
+      context: ./ninjareact
+    container_name: django_backend
+    command: python manage.py runserver 0.0.0.0:8000
+    ports:
+      - "8000:8000"
+    volumes:
+      - ./ninjareact:/app
```
Comment (joefutrelle) [inline docker-compose.yml:10]: application code is included in the Docker image  by your Dockerfile and should not be mapped into the container at runtime.

**Review by joefutrelle — CHANGES_REQUESTED** [review]
Thanks for the PR! I see numerous issues with it that will require some restructuring.

**Comment by joefutrelle** [comment]
@cpwallah I think there are a few ways we could complete this task:

1. You make the requested changes and I will review them;
1. I take over this task and commit to this branch;
1. I take over this task on a new branch / PR

My preference is the latter option given the amount of restructuring required in this PR.

**Comment by cpwallah** [comment]
i will try to resolve these issue

**Comment by joefutrelle** [comment]
@cpwallah I'm also bringing in a second reviewer, @mike-kaimika

**Comment by cpwallah** [comment]
okay @joefutrelle as i am new in using docker would you guide me if needed?

**Comment by joefutrelle** [comment]
> okay @joefutrelle as i am new in using docker would you guide me if needed?

If you're new to using Docker then you may want to find a project outside of this organization that you can contribute to. If you're interested in open-source development and Docker, then feel free to follow the linked issue #5 which will be associated with an upcoming PR from me or @mike-kaimika completing the work.

**Comment by cpwallah** [comment]
hello plz repopen  it @joefutrelle

**Comment by cpwallah** [comment]
now everything is running fine as i have set up it

**Pull request closed without merging**